src/types/label.js

```diff
@@ -0,0 +1,106 @@
+import {drawBox, drawLabel, measureLabelSize, isLabelVisible, getChartPoint, getBoxCenterPoint} from '../helpers';
+import {Element} from 'chart.js';
+
+export default class LabelAnnotation extends Element {
+
+  inRange(mouseX, mouseY) {
+    const {x, y, width, height} = this.getProps(['x', 'y', 'width', 'height']);
+
+    return this.labelTextRect && mouseX >= x &&
+			mouseX <= x + width &&
+			mouseY >= y &&
+			mouseY <= y + height;
+  }
+
+  getCenterPoint(useFinalPosition) {
+    return getBoxCenterPoint(this, useFinalPosition);
+  }
+
+  draw(ctx) {
+    if (this.labelTextRect) {
+      ctx.save();
+      drawBox(ctx, this, this.options);
+      drawLabel(ctx, this.labelTextRect, this.options);
+      ctx.restore();
+    }
+  }
+
+  resolveElementProperties(chart, options) {
+    const point = getChartPoint(chart, options);
+
+    if (isLabelVisible(options)) {
+      const labelSize = measureLabelSize(chart.ctx, options);
+      const elemDim = measureDimension(point, labelSize, options);
+      this.labelTextRect = {

```

Comment:
> I think I'd call it `this.labelRect` for consistency

@kurkle I purposely didn't use `labelRect` because the same attribute is used in line annotation but with different "meaning" (it's not ONLY the rect of the text but rect + padding and borderwidth). Here it is only the rect of the text.

Of course I can change but we have to pay attention that `label.labelRect` and `line.labelRect `contain different things